presto-main/src/main/java/com/facebook/presto/execution/buffer/OutputBuffer.java

```diff
@@ -111,6 +113,12 @@
      */
     void fail();
 
+    void setNoMorePagesForLifespan(Lifespan lifespan);
+
+    void registerLifespanFinishCallback(Consumer<Lifespan> callback);
+
+    boolean isLifespanFinished(Lifespan lifespan);
```

Comment:
It sounds weird to ask an `OutputBuffer` whether a lifespan is finished with `isLifespanFinished`. I think what we are really asking is "has all the output pages of this lifespan been acked?", and we should name this method accordingly.

Reply:
@nezihyigitbasi : I see your point. The name is decided after non-lifespan version API (`isFinished`). But I do agree with you now with `lifespan`, it sounds more confusing. 

I will think about it. Maybe `isFinishedForLifespan` to make the comparison with `isFinished` more obvious,  and add some docs.